app/models/supermarket.rb

```diff
@@ -1,3 +1,10 @@
 class Supermarket < ApplicationRecord
   has_many :items
-end
\ No newline at end of file
+  has_many :customers, through: :items
+  validates :name, presence: true
+  validates :location, presence: true
+
+  def market_customers
+    Supermarket.joins(items: :customers).where("supermarkets.name = ?", self.name).distinct.pluck("customers.name")
```

Comment:
Nice job using a `.joins` here. If you look at line 3, that association will allow you to start right at `customers` so you might be able to just do a `customers.distinct`.